core/src/commonMain/kotlin/com/xebia/functional/xef/llm/Chat.kt

```diff
@@ -102,6 +102,15 @@ interface Chat : LLM {
     return promptMessages(prompt.toMessages(), scope, functions, promptConfiguration)
   }
 
+  @AiDsl
+  suspend fun promptMessage(
```

Comment:
Should we use this method for the `promptMessage(question: String, ....)` function in line 78?

Something like this

```kotlin
  @AiDsl
  suspend fun promptMessage(
    question: String,
    scope: Conversation,
    promptConfiguration: PromptConfiguration = PromptConfiguration.DEFAULTS
  ): String =
    promptMessage(question.toMessages(), scope, promptConfiguration)
```

And, maybe we should also create  `fun promptMessage(prompt: Propmt,...)` and we will have the 3 implementations in the same way that the plural functions `promptMessages`

Reply:
To do this we need a form of template separator in the string such as:
```txt
system:
  This is valid yaml parseable as a system message
user: 
  This goes to a user message  
```
Otherwise, it all shows up as a user message, which is the same as we have now.
I think this makes sense to implement when we have Prompt persistence and loading prompts by ID.

A user wants a prompt to persist and edit it as a whole but the structure should remain so we can always go back to the list of messages.

Reply:
I also added `promptMessage` which returns the single String response but takes a `List<Message>`

Reply:
I think I missunderstood your original question and gave you instead an answer on how I was thinking on persisting templates as String, yes we could use the promptMessage method but since that one currently returns a String and not a serialized `A` which is what we need here with precision to clean up and parse the previous input. 
We could introduce a version of serializable prompts that take a List of messages and refactor wherever makes sense.